tests/test_preprocessing.py

```diff
@@ -146,3 +150,42 @@ def test_make_dataset_from_sgf(self):
                 -1
             )]
         self.assertEqualData(expected_data, recovered_data)
+
+
+    def test_rotate_pyfunc(self):
+        def test_symmetry(s, i, r1, r2):
+            # features
+            x = (r1[0][i] == symmetries.apply_symmetry_feat(sym, r2[0][i])).all()
+            # pi
+            pi = (r1[1][i] == symmetries.apply_symmetry_pi(sym, r2[1][i])).all()
+            # value
+            v = (r1[2][i] == r2[2][i]).all()
+            return x and pi and v
+
+        num_records = 200
```

Comment:
I mention this only because I want to link to the math problem but this code has ~1/1e10 chance of failing
see https://en.wikipedia.org/wiki/Coupon_collector%27s_problem

Reply:
can you use tf.set_random_seed to make the test deterministic?

Reply:
(tf.set_random_seed and in conjunction with random.seed() )

Reply:
I'll try but ugh.